src/components/VmStatus/VmStatus.js

```diff
@@ -0,0 +1,118 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import { Link } from 'react-router-dom';
+
+import {
+  VM_STATUS_ERROR,
+  VM_STATUS_STARTING,
+  VM_STATUS_RUNNING,
+  VM_STATUS_OFF,
+  VM_STATUS_POD_ERROR,
+  VM_STATUS_UNKNOWN,
+} from '../../constants';
+
+import { getSubPagePath } from '../../utils';
+import { PodModel, VirtualMachineModel } from '../../models';
+
+export const getVmStatusDetail = (vm, pod) => {
+  if (!get(vm, 'spec.running', false)) {
+    return { status: VM_STATUS_OFF };
+  }
+
+  if (get(vm, 'status.ready', false)) {
+    // we are all set
+    return { status: VM_STATUS_RUNNING };
+  }
+
+  if (get(vm, 'status.created', false)) {
+    // created but not yet ready
+    if (pod) {
+      // pod created, so check for it's potential error
+      const notReadyCondition = get(pod, 'status.conditions', []).find(condition => condition.status !== 'True');
+      if (notReadyCondition) {
+        // at least one pod condition not met, let the user analyze issue via Pod events
+        return { status: VM_STATUS_POD_ERROR, message: notReadyCondition.message };
```

Comment:
Can we also include check for VMI errors ?

Reply:
My thinking here is, if the VM `status.ready` is`true`, the `VMI` should not report any error.

And as the user is interested in just a limited (means "status") information, we do not need to complicate things by adding too many details.
If a VM reports an error, the UI leads the user for appropriate pages to resolve/analyze.